### ISSUE TITLE
Implement a new UX for filling tax forms on Open Collective

### ISSUE THREAD

**Issue by Betree** [opencollective/opencollective#7218]
Part of [[Project] Internalize the tax forms process #7216](https://github.com/opencollective/opencollective/issues/7216)

# The current flow

<details>

<summary>Filling in a tax form</summary>

1. Expense payee submits an expense and gets a warning telling him an email to fill in a tax form 
![image](https://github.com/opencollective/opencollective/assets/1556356/5d7de116-cd72-4912-bbfa-a3c0707cb6d7)
2. User clicks on the link in the email
![image](https://github.com/opencollective/opencollective/assets/1556356/0bdbded3-58d6-40d6-8486-255eb4a087d7)
3. The user fills in the tax form fields, which adapt themselves based on some custom inputs
![image](https://github.com/opencollective/opencollective/assets/1556356/59ca53ce-d6b8-4feb-9b1b-5935ec6801ff)
4. The user can preview the document, then submits it by clicking "I agree"
![image](https://github.com/opencollective/opencollective/assets/1556356/319d25c2-c31a-4866-b295-ffd7f2673ab8)
5. User gets a confirmation by email with the PDF attached. The expense is "unlocked" and will be paid by the host. 
![image](https://github.com/opencollective/opencollective/assets/1556356/ef64a476-0b19-4f56-9729-31b7f720f905)

</details>

# Goals of the new flow

- Implement the same form logic we have on Dropbox forms in our own interface
- [Preview/generate the PDFs](https://github.com/opencollective/opencollective/issues/7217) (W9, W8 or W8-BEN) directly in the flow
- Adapt the existing flow to direct users to this new form in a friendly way
- Ensure we respect https://github.com/opencollective/opencollective/issues/7220 (use the same language as the existing form, collect the signature explicitly, ask for 2FA).

**Comment by Memo-Es** [comment]
https://www.loom.com/share/b5ceb31431324211b3d44be1dab70a0d?sid=4513009a-e943-44b5-928e-65d669ec4b30

@Betree

**Comment by Betree** [comment]
@Memo-Es Adding the form in the settings as proposed in this design makes sense. Adding the ability to update your tax information is a nice touch (that was previously not possible with Dropbox Forms), but it creates two additional challenges that we'll need to address:
1. Tax forms have an expiry date: once generated, they're valid for 3 years. According to https://github.com/opencollective/opencollective/issues/6649, we don't want to keep storing the info after this delay. So if an account is still submitting expenses with tax information that's about to expire (i.e., the tax info is 2 years old), we should have an email notification + a warning on the form asking them to re-confirm their info.
2. We don't want people to update their tax information for already paid expenses. That means we shouldn't override past tax information and make sure that expenses are tied to a specific entry in `LegalDocuments` (the table where we store tax information).

As a simplification, I would keep the updating part out of the scope of the initial implementation. We can eventually add a message like `If your tax information needs to be updated, please contact support`.

**Comment by Betree** [comment]
Some constraints we'd like to enforce in the form, based on this year's feedback:
- Names should be collected using 3 separate fields: First/Middle/Last Name
- For "Status" use a select to try and enforce types (LLC, C-Corp, etc). "Other" should be an option and will pop up a string field allowing custom values.
- US Addresses are expected to be collected as:
  - Address Line 1
  - Address Line 2
  - City
  - State
  - Zip
  - Country
- We should collect a "Main email address", to clarify who's responsible for replying if there's any enquire about the provided tax info.

**Comment by Betree** [comment]
**API requirements** _(to be updated)_
- Add a `requiredLegalDocuments` field on `Account` (to know if the account needs to complete a tax form).

**Comment by Betree** [comment]
@Memo-Es The Figma link in [your comment](https://github.com/opencollective/opencollective/issues/7218#issuecomment-1916125596) does not work anymore, I was not able to find the design on Figma (both Lab and Production ready). 

I remember you showing the signature as part of a modal. I'm not sure why we had that in a separate modal; for now I'm going with the same solution as Dropbox Forms, which is to show it directly in the form:

![image](https://github.com/opencollective/opencollective/assets/1556356/85acbf34-18f5-429b-9f24-7a24691c70fb)

**Comment by Memo-Es** [comment]
@Betree it was in the production-ready file before yesterday:
![image](https://github.com/opencollective/opencollective/assets/49539954/e20440bf-3705-4b02-9f30-b05fadd33957)

I integrated the sign field in the form as you suggested:
![image](https://github.com/opencollective/opencollective/assets/49539954/d5bfa8c0-e5ec-44c4-9b77-e153ccdfdc85)

I also tagged you in a comment.

Do let me know if there is any design input still :)

**Comment by Memo-Es** [comment]
@Betree 

https://www.loom.com/share/4c78ff9109ad420bafaa89f4baf5ac53?sid=ef8a7cc9-5bb3-4003-81d0-66d253b36f22

**Comment by Betree** [comment]
After discussing with @znarf, we've decided to remove the "Current Tax Info" button that lets users download a copy of their own PDF tax form from the dashboard. Here are the reasons:
- Payees already receive a copy of their PDF tax form by email
- We haven't got any requests for this feature; it just came out as a "nice to have" in the design
- Even though the download endpoint is protected with 2FA, not everyone has 2FA enabled, and it dangerously increases the risk of data theft (this form contains addresses, SSN, tax IDs, ...etc)

In summary, high risk VS unclear benefit.

We still want to allow the download of PDFs for host admins, but we can force them to have 2FA enforced - which will greatly limit the risk.
